source/docs/software/wpilib-tools/robot-characterization/analyzing-feedback.rst

```diff
@@ -0,0 +1,81 @@
+.. image:: images/selectingunits.png
+   :alt: Picture of selecting units dropdown
+
+Additionally, if your choice of units requires a physical measurement (i.e. it is non-rotational), it is also important to specify the relevant mechanism dimension used for conversion.
+
+.. image:: images/wheeldiameter.png
+   :alt: Setting wheel diameter in specified units
+
+Enter Controller Parameters
+---------------------------
+
+The calculated feedforward gains are *dimensioned quantities*. Unfortunately, not much attention is often paid to the units of PID gains in FRC controls, and so the various typical options for PID controller implementations differ in their unit conventions (which are often not made clear to the user).
+
+To specify the correct settings for your PID controller, use the following options.
+
+.. image:: images/controllersettings.png
+   :alt: Picture of the controller settings
+
+- **Gain Settings Preset:** This drop-down menu will auto-populate the remaining fields with likely settings for one of a number of common FRC controller setups. Note that some settings, such as post-encoder gearing, PPR, and the presense of a slave motor must still be manually specified (as the analyzer has no way of knowing these without user input), and that others may vary from the given defaults depending on user setup.
+- **Controller Period:** This is the execution period of the control loop, in seconds. The default RIO loop rate is 50Hz, corresponding to a period of .02s. The onboard controllers on most "smart controllers" run at 1Khz, or a period of .001s.
+- **Max Controller Output:** This is the maximum value of the controller output, with respect to the PID calculation. Most controllers calculate outputs with a maximum value of 1, but early versions of the Talon firmware have a maximum output of 1023.
+- **Time-Normalized Controller:** This specifies whether the PID calculation is normalized to the period of execution, which affects the scaling of the D gain.
+- **Controller Type:** This specifies whether the controller is an onboard RIO loop, or is running on a smart motor controller such as a Talon or a SPARK Max.
+- **Post-Encoder Gearing:** This specifies the gearing between the encoder and the mechanism itself. This is necessary for control loops that do not allow user-specified unit scaling in their PID computations (e.g. those running on Talons). This will be disabled if not relevant.
+- **Encoder PPR:** This specifies the pulses-per-revolution (not cycles per revolution) of the encoder used, which is needed in the same caes as Post-Encoder Gearing.
+- **Has Slave:** Whether there is a motor controller slaved to the controller running the control loop, if the control loop is being run on a peripheral device. This changes the effective loop period.
+- **Slave Update Period:** The rate at which the slave (if present) is updated. By default, this is 100Hz (ever .01s) for Talons and Sparks, but can be changed.
+
+Entering Optimal Constraints
```

Comment:
Weights is a more accurate term, but that doesn't feel right either (the weights are actually `1/q^2` where q is the tolerance). Maybe "Entering state and input tolerances"? If they were constraints, they'd get violated all the time.

Reply:
They're "constraints" on the optimality criterion, not on the values.  I'm rewording this to be a bit clearer.

Reply:
Even that is inaccurate imo. The optimality criterion is `min_u J = \int_0^\infty (x^T Q x + u^T R u) dt s.t. dx/dt = Ax + Bu`. Q and its elements are not constraints, they are weights of the cost function. In controls nomenclature, `dx/dt = Ax + Bu` is a constraint.